src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -5,171 +5,113 @@
+        path.dirname(filepath),
+        'package.json'
+    )
+    if (!packageJsonPath) {
+        throw new Error( // TODO: Do we want to localize errors? This might be confusing if we need to review logs.
+            localize(
+                'AWS.error.sam.local.package_json_not_found',
+                'Unable to find package.json related to {0}',
+                filepath
             )
-            const workspaceFolder:
-                vscode.WorkspaceFolder | undefined = vscode.workspace.getWorkspaceFolder(document.uri)
-
-            if (!workspaceFolder) {
-                throw new Error(`Source file ${document.uri} is external to the current workspace.`)
-            }
-
-            lenses.push(this.generateLocalInvokeCodeLens(document, range, handler.handlerName, false, workspaceFolder))
-            lenses.push(this.generateLocalInvokeCodeLens(document, range, handler.handlerName, true, workspaceFolder))
+        )
+    }
 
-            try {
-                lenses.push(this.generateConfigureCodeLens(document, range, handler.handlerName, workspaceFolder))
-            } catch (err) {
-                const error = err as Error
+    return path.dirname(packageJsonPath)
+}
 
-                logger.error(
-                    `Could not generate 'configure' code lens for handler '${handler.handlerName}': `, error
-                )
+export function initialize({
+    configuration,
+    toolkitOutputChannel,
+    processInvoker = new DefaultSamCliProcessInvoker(),
+    taskInvoker = new DefaultSamCliTaskInvoker()
+}: CodeLensProviderParams): void {
+    vscode.commands.registerCommand(
+        getInvokeCmdKey('javascript'),
+        async (args: LambdaLocalInvokeArguments) => {
+
+            const activeFilePath = args.document.uri.fsPath
+            if (!activeFilePath) { // Should we log a warning or throw an error?
+                throw new Error("'vscode.window.activeTextEditor' not defined")
+            }
+            const samProjectCodeRoot = await getSamProjectDirPathForFile(activeFilePath)
+            const debugConfig: NodeDebugConfiguration = {
+                type: 'node',
+                request: 'attach',
+                name: 'SamLocalDebug',
+                preLaunchTask: undefined,
+                address: 'localhost',
+                port: await getDebugPort(),
+                localRoot: samProjectCodeRoot,
+                remoteRoot: '/var/task',
+                protocol: 'inspector',
+                skipFiles: [
+                    '/var/runtime/node_modules/**/*.js',
+                    '<node_internals>/**/*.js'
+                ]
             }
-        })
-
-        return lenses
-    }
 
-    public resolveCodeLens(
-        codeLens: vscode.CodeLens,
-        token: vscode.CancellationToken
-    ): vscode.ProviderResult<vscode.CodeLens> {
-        throw new Error('not implemented')
-    }
+            if (args.debug) {
```

Comment:
if statement not required due to line 69?

Reply:
Good callout. The `port`/`debugPort`/`debug` was a bit sloppy. Removed `debugPort` which was a duplicate of the `DebugConfiguration.port`. Changed `debug` to `isDebug`.